src/scripts/main.js

```diff
@@ -1,3 +1,313 @@
+
+          resolve();
+        }, 100);
+      });
+
+      await doStep;
+    }
+
+    listCells[oldCellIndex - (oneStep * stepCount)].classList
+      .remove(`field-cell--move-${sideMove}--${didMerge
+        ? stepCount + 1 : stepCount}`);
+  }
+
+  didMerge && merge(oldCellIndex - (oneStep * (stepCount + 1)),
+    oldCellIndex - (stepCount && stepCount * oneStep));
+};
+
+const merge = async(newCellIndex, oldCellIndex) => {
+  const newNumber = +listCells[oldCellIndex].textContent * 2;
+
+  if (newNumber === 2048) {
+    winMessage.classList.remove('hidden');
+  }
+
+  listCells[newCellIndex]
+    .classList = (`innerField field-cell--${newNumber} field-cell--merge`);
+  listCells[newCellIndex].textContent = newNumber;
+
+  listCells[oldCellIndex].classList = 'innerField';
+  listCells[oldCellIndex].textContent = '';
+
+  const cancelClass = new Promise((resolve) => {
+    window.setTimeout(() => {
+      listCells[newCellIndex].classList.remove('field-cell--merge');
+      resolve();
+    }, 150);
+  });
+
+  await cancelClass;
+};
+
+const lose = (changeConditions) => {
+  const left = changeConditions('left')[1]
+  || changeConditions('left')[2] || changeConditions('left')[3];
+
+  const right = changeConditions('right')[1]
+  || changeConditions('right')[2] || changeConditions('right')[3];
+
+  const up = changeConditions('up')[1]
+  || changeConditions('up')[2] || changeConditions('up')[3];
+
+  const down = changeConditions('down')[1]
+  || changeConditions('down')[2] || changeConditions('down')[3];
+
+  if (findTakenCells().length === 16 && !left && !right && !up && !down) {
+    loseMessage.classList.remove('hidden');
+  }
+};
+
+const moveCells = async(sideMove) => {
+  const doMove = new Promise(async(resolve) => {
+    isMovingNow = true;
+
+    const takenCells = findTakenCells();
+    let didMove = false;
+    const isDescCycle = (sideMove === 'right' || sideMove === 'down') && true;
+
+    for (let cellIndex = isDescCycle ? (takenCells.length - 1) : 0;
+      cellIndex >= 0 && cellIndex <= takenCells.length - 1;
+      isDescCycle ? cellIndex-- : cellIndex++) {
+      let stepCount = 0;
+      let didMerge = false;
+
+      const changeConditions = (moveSide) => {
+        let step = 1;
+        let firstCondition = (takenCells[cellIndex] % 4 === 1) && true;
+        let secondCondition = (takenCells[cellIndex] % 4 === 2) && true;
+        let thirdCondition = (takenCells[cellIndex] % 4 === 3) && true;
+
+        if (moveSide === 'up') {
+          firstCondition = takenCells[cellIndex] >= 4
+          && takenCells[cellIndex] <= 7;
+
+          secondCondition = takenCells[cellIndex] >= 8
+          && takenCells[cellIndex] <= 11;
+
+          thirdCondition = takenCells[cellIndex] >= 12
+          && takenCells[cellIndex] <= 15;
+          step = 4;
+        }
+
+        if (moveSide === 'right') {
+          step = -1;
+
+          firstCondition = (takenCells[cellIndex] % 4 === 2) && true;
```

Comment:
You don't need` && true`, because it won't influence result.
`false && true` will give `false`
`true && true` will give `true`
So your variable only depends on `takenCells[cellIndex] % 4 === 1`
Also check the other cases